tomtom-navigation-core-examples/usecase/src/main/kotlin/com/example/usecase/MainViewModel.kt

```diff
@@ -0,0 +1,264 @@
+    }
+
+    /**
+     * The RoutePlanningCallback itself has three methods.
+     * - The `onFailure()` method is triggered if the request fails.
+     * - The `onSuccess()` method returns RoutePlanningResponse containing the routing results.
+     * - The `onRoutePlanned()` method is triggered when each route is successfully calculated.
+     *
+     * This example draws the first retrieved route on the map.
+     * You can show the overview of the added routes using the TomTomMap.zoomToRoutes(Int) method.
+     * Note that its padding parameter is expressed in pixels.
+     */
+    private val routePlanningCallback = object : RoutePlanningCallback {
+        override fun onSuccess(result: RoutePlanningResponse) {
+            val routePlan = RoutePlan(result.routes)
+            navigationVisualization.displayRoutePlan(routePlan)
+            navigationVisualization.selectRoute(routePlan.routes.first().id)
+        }
+
+        override fun onFailure(failure: RoutingFailure) {
+            _routingFailure.value = failure
+        }
+
+        override fun onRoutePlanned(route: Route) = Unit
+    }
+
+    /**
+     * Use the SimulationLocationProvider for testing purposes.
+     */
+    private fun setSimulationLocationProviderToNavigation(route: Route) {
+        val routeGeoLocations = route.geometry.map { GeoLocation(it) }
+        val simulationStrategy = InterpolationStrategy(routeGeoLocations)
+        val oldNavigationLocationProvider = navigationLocationProvider
+        navigationLocationProvider =
+            SimulationLocationProvider.create(strategy = simulationStrategy)
+        tomTomNavigation.locationProvider = navigationLocationProvider
+        navigationLocationProvider.enable()
+        oldNavigationLocationProvider.close()
+    }
+
+    /**
+     * Once navigation is started, the camera is set to follow the user position, and the location indicator is changed to a chevron.
+     * To match raw location updates to the routes, create a LocationProvider instance using MapMatchedLocationProviderFactory and set it to the TomTomMap.
+     */
+    private fun setUpMapMatchedLocationProvider() {
+        _mapMatchedLocationProvider = MapMatchedLocationProviderFactory.create(tomTomNavigation)
+        _mapMatchedLocationProvider?.enable()
+        mapLocationProvider.disable()
+    }
+
+
+    fun navigationStopped() {
+        mapLocationProvider.enable()
+        navigationLocationProvider.disable()
+        _mapMatchedLocationProvider?.close()
+    }
+
+    /**
+     * Checks whether navigation is currently running.
+     */
+    fun isNavigationRunning(): Boolean = tomTomNavigation.navigationSnapshot != null
+
+    override fun onCleared() {
+        super.onCleared()
+        _mapMatchedLocationProvider?.close()
+        tomTomNavigation.close()
+        navigationLocationProvider.close()
+        navigationTileStore.close()
+        mapLocationProvider.close()
```

Comment:
should you not call here` navigationVisualization.clear()` also?

Reply:
Do you mean `clearRoutePlan`? I see it is called when closing the `NavigationVisualization`

Reply:
Yes i meant `clearRoutePlan()` 
Okey you are closing the component. It is fine then